blog/2019-03-14-debezium-meets-quarkus.adoc

```diff
@@ -0,0 +1,193 @@
+there was a microservice ("shipment") based on Thorntail that consumed the events produced by the Debezium connector.
+We've extended the example with a new service named "shipment-service-quarkus".
+It provides the same functionality as the "shipment-service" but is implemented as a microservice based on Quarkus instead of Thorntail.
+
+This makes the overall architecture look like so:
+
+++++
+<div class="imageblock centered-image">
+    <img src="/images/outbox_pattern_quarkus.png" style="max-width:100%; margin-bottom:20px; margin-top:20px;" class="responsive-image" alt="Outbox Pattern Overview">
+</div>
+++++
+
+To retrofit the original service into a Quarkus-based application, only a few changes were needed:
+
+ * Quarkus right now supports only MariaDB but not MySQL; hence we have included an instance of MariaDB to which the service is writing
+ * The https://javaee.github.io/jsonp/[JSON-P API] used do deserialize incoming JSON messages is not supported yet; so the code has been modified to use the Jackson API which is already is supported by Quarkus
+ * Instead of the Kafka consumer API, the https://github.com/eclipse/microprofile-reactive-messaging[Reactive Messaging API] defined by MicroProfile is used to receive messages from Apache Kafka; as an implementation of that API, the one provided by the https://github.com/smallrye/smallrye-reactive-messaging[SmallRye project] is used.
+
+While the first two steps shouldn't be needed eventually, as Quarkus gains support for more technologies and APIs over time,
+the Reactive Messaging API is a nice simplification over the polling loop in the original consumer.
+All that's needed to consume messages from a Kafka topic is to annotate a method with `@Incoming`,
+and will automatically be invoked when a new message arrives:
+
+[source,java]
+----
+@ApplicationScoped
+public class KafkaEventConsumer {
+
+    @Incoming("orders")
+    public CompletionStage<Void> onMessage(KafkaMessage<String, String> message)
+            throws IOException {
+        // handle message...
+
+        return message.ack();
+    }
+}
+----
+
+The "orders" message source is configured via the MicroProfile Config API,
+which resolves it to the "OrderEvents" topic already known from the original outbox example.
+
+== Build Process
+
+The build process is almost the same as it was before.
+Besides adding the Quarkus Maven plug-in itself, only two changes were needed:
+
+ * A new build profile named `native` has been added; this is used to compile the service into a native binary image using the Quarkus Maven plug-in
+ * the `native-image.docker-build` system property is enabled when running the build; this means that the native image build is done inside of a Docker container, so that GraalVM doesn't have to be installed on the developer's machine
+
+All the heavy-lifting is done by the Quarkus Maven plug-in which is configured in _pom.xml_ like so:
```

Comment:
I think having a paragraph concerning the Quarkus extensions you use would be nice.

Reply:
@gsmet, can you elaborate, what would that entail? Maybe you can gist what you had in mind?

Reply:
Ah, never mind; I misread at first. I know what it means. Thanks :)